ruby/ql/lib/codeql/ruby/frameworks/core/Array.qll

```diff
@@ -741,19 +741,14 @@ module Array {
 
     override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
       (
-        input = "Argument[self].ArrayElement[?]" and
-        output = ["ReturnValue", "Argument[self].ArrayElement[?]"]
-        or
-        exists(ArrayIndex j | input = "Argument[self].ArrayElement[" + j + "]" |
-          j = i and output = "ReturnValue"
-          or
-          j != i and output = "Argument[self].ArrayElement[" + j + "]"
-        )
+        input = "Argument[self].ArrayElement[?," + i + "]" and
```

Comment:
Cool, I didn't know we supported syntax like `[?,0]`.

Reply:
That's also a fairly recent addition thanks to @asgerf .